More details for ARMv7 (#4693)
OttoWinter/esphomedocs: changelog/2025.2.0.rst

encourage all developers to migrate their custom components to :doc:`/components/external_components`. Please see
:ref:`a_note_about_custom_components` for a detailed explanation and next steps.

> Removal of ``armv7`` Docker (and HA add-on) Support

Reviewer: We shouldn't be abbreviating Home Assistant as `HA`

---------------------------------------------------

> ESPHome no longer supports ``armv7`` (32-bit ARM) hardware. This is due to both waning support as it relates to tooling and

Reviewer: 32-bits ARM is not just `armv7`, so this addition seem incorrect.

Author: Assuming we're not also providing `armhf`, then it's correct.  And it's more clear for users that don't understand what `armv7` is.  The 64-bit platforms actually have `64` in the name.

Reviewer: Sure, but there are multiple 32 bits platforms for 32 bits.... 🤷

Author: But are we providing any of them?

Reviewer: from the add-on perspetive, we could, yes; so we should be clear IMHO.

Reviewer 2: Maybe this?

```suggestion
ESPHome no longer supports the 32-bit  ``armv7`` architecture. This is due to both waning support as it relates to tooling and
```

performance reasons. We strongly recommend moving to a more modern architecture, especially if you're using the ESPHome
Device Builder to build/compile firmware for your devices.

> `More detailed explanation. <https://github.com/esphome/issues/issues/6770#issuecomment-2672644628>`__

Reviewer: It shouldn't be needed to link issues. If there is missing context here, we should add it. Driving users into technical issues isn't something we should aim for.

Author: Do you want me to copy that section into here?  I was trying to avoid bloating the changelog needlessly.

Reviewer: That is not what I'm suggesting. I'm suggesting that if the context is incomplete, we need to complete it.

encourage all developers to migrate their custom components to :doc:`/components/external_components`. Please see
:ref:`a_note_about_custom_components` for a detailed explanation and next steps.

> Removal of ``armv7`` Docker and Home Assistant add-on support

Reviewer 3: ```suggestion
Removal of ``armv7`` Docker (incl. Home Assistant add-on support for ``armv7``) 
```
Title looks like HA add-on is completely unsupported.

Reviewer: Yeah agreed with that comment. The other approach would be just:

```suggestion
Removal of ``armv7`` support
```

Simple, to the point, applies everywhere.

Author: If you're going to change the title text, then you need change the underline too.  Otherwise, it can't be trivially committed. :-)

> ---------------------------------------------------------------

Reviewer 2: ```suggestion
-------------------------------------------------------------
```

encourage all developers to migrate their custom components to :doc:`/components/external_components`. Please see
:ref:`a_note_about_custom_components` for a detailed explanation and next steps.

> Removal of ``armv7`` Docker (and Home Assistant add-on) Support

Reviewer: ```suggestion
Removal of ``armv7`` Docker and Home Assistant add-on support
```
